sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RegisterAppInterface.java

```diff
@@ -573,32 +582,66 @@ public String getHashID() {
    
     public void setHashID(String hashID) {
 		setParameters(KEY_HASH_ID, hashID);
-    }        
-    
+    }
+
 	/**
 	 * Gets the unique ID, which an app will be given when approved
-	 * 
+	 *
 	 * @return String - a String value representing the unique ID, which an app
 	 *         will be given when approved
 	 * @since SmartDeviceLink 2.0
 	 */
-    public String getAppID() {
-        return getString(KEY_APP_ID);
-    }
+	public String getAppID() {
+		return getString(KEY_APP_ID);
+	}
 
 	/**
 	 * Sets a unique ID, which an app will be given when approved
-	 * 
+	 *
 	 * @param appID
 	 *            a String value representing a unique ID, which an app will be
 	 *            given when approved
 	 *            <p></p>
 	 *            <b>Notes: </b>Maxlength = 100
 	 * @since SmartDeviceLink 2.0
 	 */
-    public void setAppID(@NonNull String appID) {
+	public void setAppID(@NonNull String appID) {
 		setParameters(KEY_APP_ID, appID);
-    }
+	}
+
+	/**
+	 * Gets the unique ID, which an app will be given when approved
+	 *
+	 * @return String - a String value representing the unique ID, which an app
+	 *         will be given when approved
+	 * @since SmartDeviceLink 4.6
+	 */
+	public String getFullAppID() {
+		return getString(KEY_FULL_APP_ID);
+	}
+
+	/**
+	 * Sets a unique ID, which an app will be given when approved
+	 *
+	 * @param fullAppID
+	 *            a String value representing a unique ID, which an app will be
+	 *            given when approved
+	 *            <p></p>
+	 *            <b>Notes: </b>Maxlength = 100
+	 * @since SmartDeviceLink 2.0
+	 */
+	public void setFullAppID(@NonNull String fullAppID) {
+		if (fullAppID != null) {
+			setParameters(KEY_FULL_APP_ID, fullAppID);
+			String appID;
+			if (fullAppID.length() <= APP_ID_MAX_LENGTH) {
+				appID = fullAppID;
+			} else {
+				appID = fullAppID.replace("-", "").substring(0, APP_ID_MAX_LENGTH);

```

Comment:
Theres nothing in the proposal about modifying the app id should it be > than the max length. removing `-` might break comparisons (the previous block keeps the `-`'s. Also, I don't believe they even define a max length.